### PR TITLE
Init Logger before Configs

### PR DESCRIPTION
**Pull request by Spellchaser**
The project attempts to use the logger before it's initialized. 
For example, it would never print out log.Fatal(4, "error with configuration file: %s", err) and merely exit. Additionally the logger is used in loading configs meaning they silently fail. 

The user specified logging levels are applied after configs are parsed.

### CODE REVIEW

**Comment by Dieterbe** [comment]
> For example, it would never print out log.Fatal(4, "error with configuration file: %s", err) and merely exit. 

omg you're right. how silly.
```
$ ./build/metrictank -config build/metrictank; echo $?
1
```

this will need fixing though, see the red X to see build errors.
probably also cleaner to just amend this batch of 6 commit into 1 commit. (i can also squash when merging)

File: metrictank.go [hunk 1]
```diff
@@ -175,6 +176,9 @@ func main() {
 	case 6:
 		log.Level(log.FATAL)
 	}
+	mdata.LogLevel = logLevel
+	inKafkaMdm.LogLevel = logLevel
+	api.LogLevel = logLevel
```
Comment (Dieterbe) [inline metrictank.go]: why did you move the assignments down instead of just keeping them as is?

Comment (Spellchaser) [reply metrictank.go]: Honestly, a testing artifact. I did think that since we're initializing the logger first, we'd want to configure it first.

**Comment by Spellchaser** [comment]
T'was just gofmt. Hmm, I must have messed up the workflow since it's our master branch. If you could squash the commits when merging, that'd be easiest :)

File: metrictank.go [hunk 2]
```diff
@@ -102,6 +102,11 @@ func init() {
 func main() {
 	startupTime = time.Now()
 
+	/***********************************
+		Initialize Logger
+	***********************************/
+	log.NewLogger(0, "console", fmt.Sprintf(`{"level": %d, "formatting":false}`, logLevel))
```
Comment (Dieterbe) [inline metrictank.go]: doesn't this have to come after the flag.Parse call though?

Comment (Spellchaser) [reply metrictank.go]: It definitely should if the logLevel specified mattered. But as you said in the commented issue #4055, it doesn't actually affect the log levels until they're set later on.

I agree, it should probably go after the version check.

**Comment by Spellchaser** [comment]
Lmao that didn't work.
